core/src/test/java/io/grpc/internal/ManagedChannelImplIdlenessTest.java

```diff
@@ -194,6 +195,63 @@ public ClientTransport answer(InvocationOnMock invocation) throws Throwable {
   }
 
   @Test
+  public void newCallResetsGracePeriod() throws Exception {
+    final EquivalentAddressGroup addressGroup = addressGroupList.get(1);
+    doAnswer(new Answer<ClientTransport>() {
+        @Override
+        public ClientTransport answer(InvocationOnMock invocation) throws Throwable {
+          return channel.tm.getTransport(addressGroup);
+        }
+      }).when(mockLoadBalancer).pickTransport(any(Attributes.class));
+
+    ClientCall<String, Integer> call = channel.newCall(method, CallOptions.DEFAULT);
+    call.start(mockCallListener, new Metadata());
+    call.cancel("cleanup", null);
+    executor.runDueTasks();
+
+    timer.runDueTasks();
+    verify(mockLoadBalancerFactory).newLoadBalancer(anyString(), same(channel.tm));
+    verify(mockLoadBalancer).pickTransport(any(Attributes.class));
+
+    // Enter grace period
+    timer.forwardTime(TimeUnit.SECONDS.toMillis(IDLE_TIMEOUT_SECONDS)
+        - ManagedChannelImpl.IDLE_GRACE_PERIOD_MILLIS, TimeUnit.MILLISECONDS);
+
+    call = channel.newCall(method, CallOptions.DEFAULT);
+    call.start(mockCallListener, new Metadata());
+    call.cancel("cleanup", null);
+    executor.runDueTasks();
+
+    // Load balancer was reused.
+    timer.runDueTasks();
```

Comment:
assertEquals(1, timer.runDueTasks()) to verity that the idle timer is actually run?